bundle/config/resources_test.go

```diff
@@ -125,3 +127,55 @@ func TestVerifySafeMergeForRegisteredModels(t *testing.T) {
 	err := r.VerifySafeMerge(&other)
 	assert.ErrorContains(t, err, "multiple resources named bar (registered_model at bar.yml, registered_model at bar2.yml)")
 }
+
+// This test ensures that all resources have a custom marshaller and unmarshaller.
+// This is required because DABs resources map to Databricks APIs, and they do so
+// by embedding the corresponding Go SDK structs.
+// The Go SDK structs implement custom marshalling and unmarshalling methods. If we
```

Comment:
**Some** Go SDK structs implement the custom marshaller. It is only needed for CLI to implement it if the corresponding Go SDK struct also implements it. Over-implemting should not have any impact, thought.

Reply:
Thanks! Modified the comment here.